packages/flutter_tools/test/flutter_build_apk.shard/flutter_build_apk_test.dart

```diff
@@ -0,0 +1,87 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io' as io;
+
+import 'package:file/file.dart';
+
+import '../integration.shard/test_utils.dart';
+import '../src/common.dart';
+
+void main() {
+  final String flutterRoot = getFlutterRoot();
+  final String flutterBin = fileSystem.path.join(flutterRoot, 'bin', 'flutter');
+
+  late Directory tmpDir;
+
+  setUp(() {
+    tmpDir = fileSystem.systemTempDirectory.createTempSync();
+  });
+
+  tearDown(() {
+    tryToDelete(tmpDir);
+  });
+
+  Future<void> setGradleLoggingLevel(
+    String level, {
+    required Directory projectDir,
+  }) async {
+    // Open gradle.properties and append to it.
+    final Directory androidDir = projectDir.childDirectory('android');
+    final File gradleDotProperties = androidDir.childFile('gradle.properties');
+    final io.IOSink sink = gradleDotProperties.openWrite(mode: FileMode.append);
+
+    sink.writeln('org.gradle.logging.level=$level');
+    await sink.flush();
+    await sink.close();
+
+    // For debugging, print the current output.
+    io.stderr.writeln('${gradleDotProperties.path}:');
+    io.stderr.writeln(gradleDotProperties.readAsStringSync());
+  }
+
+  // Normally these tests should take about a minute, but sometimes for
+  // unknown reasons they can take 30m+ and timeout. The intent behind this loop
+  // is to get more information on what exactly is happening.
+  for (int i = 1; i <= 10; i++) {
```

Comment:
SGTM. However, if these randomly long builds are only clean builds (where nothing is cached), then I don't think this loop will accomplish much. IIRC I wrote a comment somewhere about whether or not these were only happening on first builds in test—I'll go look for it.

Reply:
Nevermind, that was about Xcode builds: https://github.com/flutter/flutter/issues/157847#issuecomment-2455750155

Reply:
Regardless, as long as _some_ build takes a shockingly time to complete, then this will timeout as desired; so SGTM